### PR TITLE
[WIP] Adds ban message so users know they got banned more clearly

### PR DESCRIPTION
**Pull request by k-k-karma**
:cl: karma
add: Adds ban message so users know they got banned more clearly
/:cl:

### CODE REVIEW

File: hippiestation/code/_onclick/hud/fullscreen.dm [hunk 1]
```diff
@@ -0,0 +1,6 @@
+/obj/screen/fullscreen/ban_message
+    show_when_dead = true
+    icon = 'hippiestation/icons/mob/screen_full.dmi'
+    icon_state = "banned"
+    layer = 23 // ABOVE EVERYTHING FUCK YOU
```
Comment (yoyobatty) [inline hippiestation/code/_onclick/hud/fullscreen.dm]: Don't use this, use layer defines in `_DEFINES`

File: hippiestation/code/modules/admin/DB_ban/functions.dm [hunk 2]
```diff
@@ -0,0 +1,9 @@
+/datum/admins/DB_ban_record(bantype, mob/banned_mob, duration = -1, reason, job = "", bankey = null, banip = null, bancid = null)
+    if(ismob(banned_mob)
+        switch(bantype)
+            if(BANTYPE_TEMP)
+                banned_mob.overlay_fullscreen("ban_message", /obj/screen/fullscreen/ban_message)
+            if (BANTYPE_PERMA)
+                banned_mob.overlay_fullscreen("ban_message", /obj/screen/fullscreen/ban_message)
+    sleep(1000)//make sure it gets on the users screen
```
Comment (yoyobatty) [inline hippiestation/code/modules/admin/DB_ban/functions.dm]: what?

Comment (k-k-karma) [reply hippiestation/code/modules/admin/DB_ban/functions.dm]: i want to make sure the image is overlayed on the mob before they're kicked from the server so I added a small delay

**Comment by yskefunky** [comment]
dude im not voting on a mirror for once woa

also yes important +1

**Comment by Eagle-Eyess** [comment]
+1

File: hippiestation/code/modules/admin/DB_ban/functions.dm [hunk 3]
```diff
@@ -0,0 +1,9 @@
+/datum/admins/DB_ban_record(bantype, mob/banned_mob, duration = -1, reason, job = "", bankey = null, banip = null, bancid = null)
+    if(ismob(banned_mob))
+        switch(bantype)
+            if(BANTYPE_TEMP)
```
Comment (ghost) [inline hippiestation/code/modules/admin/DB_ban/functions.dm]:
```
if(BANTYPE_TEMP, BANTYPE_PERMA)
    ~~~
```
Also i'd run something like `banned_mob.death()` to kill them so they can't do voice activated bombs or whatever during the 10 seconds

Comment (k-k-karma) [reply hippiestation/code/modules/admin/DB_ban/functions.dm]: this runs on dead mobs but I'll lower it to 1s

File: hippiestation/code/modules/admin/DB_ban/functions.dm [hunk 4]
```diff
@@ -0,0 +1,7 @@
+/datum/admins/DB_ban_record(bantype, mob/banned_mob, duration = -1, reason, job = "", bankey = null, banip = null, bancid = null)
+    if(ismob(banned_mob))
+        switch(bantype)
+            if(BANTYPE_TEMP, BANTYPE_PERMA)
+                banned_mob.overlay_fullscreen("ban_message", /obj/screen/fullscreen/ban_message)
+    sleep(10)//make sure it gets on the users screen (10s)
```
Comment (ghost) [inline hippiestation/code/modules/admin/DB_ban/functions.dm]: outdated comment

**Comment by Kawaii-Big-Boss** [comment]
Use the adminhelp2 sound, so every time they get banned they can hear "WAAAAAH"

**Comment by t-f-w** [comment]
+1

**Comment by danhibiki** [comment]
+3 headmins and reviewed needs merge

**Comment by GuyonBroadway** [comment]
Did you test this?

**Comment by k-k-karma** [comment]
no I can't ban myself locally but I'm confident it'll work

**Comment by GuyonBroadway** [comment]
You can at least test the splash screen by calling `overlay_fullscreen("ban_message", /obj/screen/fullscreen/ban_message)` on yourself.

**Comment by k-k-karma** [comment]
im pretty sure that works too but gimmie a sec

File: hippiestation/code/modules/admin/DB_ban/functions.dm [hunk 5]
```diff
@@ -0,0 +1,7 @@
+/datum/admins/DB_ban_record(bantype, mob/banned_mob, duration = -1, reason, job = "", bankey = null, banip = null, bancid = null)
+    if(ismob(banned_mob))
+        switch(bantype)
+            if(BANTYPE_TEMP, BANTYPE_PERMA)
+                banned_mob.overlay_fullscreen("ban_message", /obj/screen/fullscreen/ban_message)
+    sleep(10)//make sure it gets on the users screen (1s)
```
Comment (ghost) [inline hippiestation/code/modules/admin/DB_ban/functions.dm]: indent this to the applying an overlay to so other bantypes don't suffer

Comment (k-k-karma) [reply hippiestation/code/modules/admin/DB_ban/functions.dm]: RESOLVED

**Comment by bardofrage** [comment]
couldn't hurt, +1.

**Comment by yobob591** [comment]
+1 dont ban me from club penguin

**Pull request merged**